packages/composites/ondo-calculated/src/lib/session.ts

```diff
@@ -0,0 +1,27 @@
+import { TZDate } from '@date-fns/tz'
+
+// Seconds relative to session boundary (-ve before, +ve after)
+export const calculateSecondsFromTransition = (
+  sessionBoundaries: string[],
+  sessionBoundariesTimeZone: string,
+) => {
+  const now = new TZDate(new Date().getTime(), sessionBoundariesTimeZone)
+
+  return sessionBoundaries.reduce((minDiff, b) => {
+    const [hour, minute] = b.split(':')
+    const session = new TZDate(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      Number(hour),
+      Number(minute),
+      0,
+      0,
+      sessionBoundariesTimeZone,
+    )
+
+    const diff = (now.getTime() - session.getTime()) / 1000
+
+    return Math.abs(diff) < Math.abs(minDiff) ? diff : minDiff
```

Comment:
Not familiar with this logic, so to double check, at 23:58, boundary 00:05 is interpreted as today 00:05 (≈ 23h53m ago) instead of tomorrow 00:05. Is that intended?